Snowflake.Data/Core/CredentialManager/Infrastructure/SFCredentialManagerInMemoryImpl.cs

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2024 Snowflake Computing Inc. All rights reserved.
+ */
+
+
+using System.Collections.Generic;
+using System.Security;
+using Snowflake.Data.Client;
+using Snowflake.Data.Core.Tools;
+using Snowflake.Data.Log;
+
+namespace Snowflake.Data.Core.CredentialManager.Infrastructure
+{
+    internal class SFCredentialManagerInMemoryImpl : ISnowflakeCredentialManager
+    {
+        private static readonly SFLogger s_logger = SFLoggerFactory.GetLogger<SFCredentialManagerInMemoryImpl>();
+
+        private Dictionary<string, SecureString> s_credentials = new Dictionary<string, SecureString>();
```

Comment:
I think we would need to think about concurrency and either to change this collection to ConcurrentDictionary or to use locks eg. ReaderWriterLockSlim. The same issue would be probably with other implementations...

Reply:
Added ReaderWriterLockSlim lock to each implementation. Thank you!!!